pkg/assetfetcher/asset.go

```diff
@@ -316,6 +316,9 @@ func (a Asset) AssetPayloadToAssetEvent() (domain.AssetEvent, error) {
 	if err != nil {
 		return domain.AssetEvent{}, err
 	}
+	if a.ID == 0 || a.IP == "" || lastScanned.IsZero() {
+		return domain.AssetEvent{}, &MissingRequiredFields{a.ID, a.IP, lastScanned}
```

Comment:
question: Is this meant to be handled in a special way somewhere else?

Reply:
Are you referring handling the empty asset or the error in a special way?
I have this ticket https://hello.atlassian.net/browse/SECD-585 to handle empty assets and make sure they're not produced. And for the error, it should bubble up and get on the error channel